Plugin is not compatible with Gradle 4.2

I get an error when trying to use Gradle 4.2 or 4.2.1 with the plugin:
```
Caused by: groovy.lang.MissingPropertyException: No such property: FOR_RUNTIME for class: org.gradle.api.attributes.Usage
        at com.novoda.gradle.release.AndroidLibrary$RuntimeUsage.getUsage(AndroidLibrary.groovy:53)
        at org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication.from(DefaultMavenPublication.java:132)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:479)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.CompositeDynamicObject.tryInvokeMethod(CompositeDynamicObject.java:98)
        at org.gradle.internal.metaobject.MixInClosurePropertiesAsMethodsDynamicObject.tryInvokeMethod(MixInClosurePropertiesAsMethodsDynamicObject.java:30)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:59)
        at com.novoda.gradle.release.ReleasePlugin$_addArtifact_closure3.doCall(ReleasePlugin.groovy:65)
        at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:71)
        at org.gradle.util.ConfigureUtil.configureTarget(ConfigureUtil.java:160)
        at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:106)
        at org.gradle.util.ConfigureUtil$1.execute(ConfigureUtil.java:123)
        at org.gradle.api.internal.AbstractPolymorphicDomainObjectContainer.create(AbstractPolymorphicDomainObjectContainer.java:61)
        at org.gradle.api.publish.internal.DefaultPublicationContainer_Decorated.create(Unknown Source)
        at org.gradle.api.publish.internal.DefaultPublicationContainer_Decorated$create.call(Unknown Source)
        at com.novoda.gradle.release.ReleasePlugin.addArtifact(ReleasePlugin.groovy:57)
        at com.novoda.gradle.release.ReleasePlugin$_attachArtifacts_closure2.doCall(ReleasePlugin.groovy:47)
        at com.novoda.gradle.release.ReleasePlugin.attachArtifacts(ReleasePlugin.groovy:45)
        at com.novoda.gradle.release.ReleasePlugin$_apply_closure1.doCall(ReleasePlugin.groovy:15)
        at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:40)
```

Looks like `FOR_RUNTIME` was removed in favour of `JAVA_RUNTIME` in [this commit](https://github.com/gradle/gradle/pull/2450/commits/8b1320fe6dccaa7cf13d25b0ca0797d5b4dd3917) from this PR: https://github.com/gradle/gradle/pull/2450

--- a/build.gradle
+++ b/build.gradle
@@ -5,5 +5,5 @@ allprojects {
         google()
         jcenter()
     }
-    version = "0.6.1"
+    version = "0.7.0"
 }

--- a/core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy
+++ b/core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy
@@ -9,19 +9,25 @@ import org.gradle.api.attributes.Usage
 import org.gradle.api.internal.DefaultDomainObjectSet
 import org.gradle.api.internal.component.SoftwareComponentInternal
 import org.gradle.api.internal.component.UsageContext
+import org.gradle.api.model.ObjectFactory
 
 class AndroidLibrary implements SoftwareComponentInternal {
 
     private final UsageContext runtimeUsage
 
     public static AndroidLibrary newInstance(Project project) {
+
+        ObjectFactory objectFactory = project.getObjects();
+        Usage usage = objectFactory.named(Usage.class, Usage.JAVA_RUNTIME);
+
+
         def configuration = project.configurations.getByName("compile")
-        return configuration ? from(configuration) : empty()
+        return configuration ? from(configuration, usage) : empty()
     }
 
-    static AndroidLibrary from(def configuration) {
-        def usage = new RuntimeUsage(configuration.dependencies)
-        new AndroidLibrary(usage)
+    static AndroidLibrary from(def configuration, Usage usage) {
+        def runtimeUsage = new RuntimeUsage(configuration.dependencies, usage)
+        new AndroidLibrary(runtimeUsage)
     }
 
     static AndroidLibrary empty() {
@@ -44,13 +50,15 @@ class AndroidLibrary implements SoftwareComponentInternal {
     private static class RuntimeUsage implements UsageContext {
 
         private final DomainObjectSet<Dependency> runtimeDependencies
+        private final Usage usage;
 
-        RuntimeUsage(DomainObjectSet<Dependency> runtimeDependencies) {
+        RuntimeUsage(DomainObjectSet<Dependency> runtimeDependencies, Usage usage) {
+            this.usage = usage;
             this.runtimeDependencies = runtimeDependencies
         }
 
         Usage getUsage() {
-            return Usage.FOR_RUNTIME
+            return usage;
         }
 
         public Set<PublishArtifact> getArtifacts() {

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.2.1-all.zip

